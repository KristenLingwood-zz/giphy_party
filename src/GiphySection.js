import React, { PureComponent } from 'react';
import axios from 'axios';
import Mosaic from './Mosaic';
import './giphySection.css'
class GiphySection extends PureComponent {
  state = {
    searchValue: '',
    giphys: [],
    loading: false
  };

  handleSearchBoxChange = (event) => {
    this.setState({ searchValue: event.target.value });
  }
  handleSearchClick = (event) => {
    event.preventDefault();
    console.log('submit!')
    this.getGiphys();
    this.setState({ ...this.state, searchValue: '' })
  }
  getGiphys = async () => {
    try {
      this.setState({ loading: true })
      const response = await axios.get(`https://api.giphy.com/v1/gifs/search`, {
        params: {
          api_key: process.env.REACT_APP_GIPHY_API_KEY,
          q: this.state.searchValue
        }
      })
      this.setState({ ...this.state, giphys: response.data.data })
    } catch (error) {
      window.alert('Sorry, there was a problem fetching giphys.', error)
    }
  }
  handleRandomClick = (event) => {
    event.target.blur();
    this.getRandomGiphys();
    this.setState({ ...this.state, searchValue: '' })
  }
  getRandomGiphys = async () => {
    try {
      this.setState({ loading: true })
      const response = await axios.get(`https://api.giphy.com/v1/gifs/random`, {
        params: {
          api_key: process.env.REACT_APP_GIPHY_API_KEY
        }
      })
      console.log("random res", response)
      this.setState({ ...this.state, giphys: [response.data.data] })
    } catch (error) {
      window.alert('Sorry, there was a problem fetching giphys.', error)
    }
  }

  render() {
    return (
      <div className="giphySection">
        <form className="form-inline" onSubmit={this.handleSearchClick}>
          <label>
            <input className="form-control" type="text" placeholder="Find me some giphys!" value={this.state.searchValue} onChange={this.handleSearchBoxChange} />
          </label>
          <input id="searchButton" className="btn btn-primary btn-sm" type="submit" value="Search" />
        </form>
        <button className="btn btn-secondary btn-sm" onClick={this.handleRandomClick}>Suprise Me</button>
        <Mosaic giphys={this.state.giphys} loading={this.state.loading} />
      </div>
    )
  }
}

export default GiphySection