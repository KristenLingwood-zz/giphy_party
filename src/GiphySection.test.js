import React from 'react';
import GiphySection from './GiphySection';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';



describe('<GiphySection />', () => {
  it('matches the snapshot', () => {
    var tree = renderer.create(<GiphySection />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});



it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<GiphySection />, div);
  ReactDOM.unmountComponentAtNode(div);
});
