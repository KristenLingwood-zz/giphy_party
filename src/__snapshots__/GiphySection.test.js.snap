// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiphySection /> matches the snapshot 1`] = `
<div
  className="giphySection"
>
  <form
    className="form-inline"
    onSubmit={[Function]}
  >
    <label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Find me some giphys!"
        type="text"
        value=""
      />
    </label>
    <input
      className="btn btn-primary btn-sm"
      id="searchButton"
      type="submit"
      value="Search"
    />
  </form>
  <form
    className="form-inline"
    onSubmit={[Function]}
  >
    <label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Enter a gif url!"
        type="text"
        value=""
      />
    </label>
    <input
      className="btn btn-primary btn-sm"
      id="submitGifButton"
      type="submit"
      value="Submit a Giphy"
    />
  </form>
  <button
    className="btn btn-secondary btn-sm"
    id="surpriseButton"
    onClick={[Function]}
  >
    Suprise Me
  </button>
  <div>
    <h3
      className="text-primary"
    >
      All the giphys say PARTY!!!
    </h3>
    <div
      className="mosaic"
    />
  </div>
</div>
`;
